src/routes.ts

```diff
@@ -21,37 +20,41 @@ router.get(
       const { cog } = req.params;
 
       const districtResponseRaw = await getDistrictFromCOG(cog);
-      const district = districtResponseRaw[0] // TODO : multiple district can have the same cog.
-      const useBanId = district?.config?.useBanId;
+      if (!districtResponseRaw.length) {
+        throw new Error(`No district found with cog ${cog}`);
+      } else if (districtResponseRaw.length > 1) {
+        throw new Error(
+          `Multiple district found with cog ${cog}. Behavior not handled yet.`
+        );
+      }
+
+      const { id, config, meta } = districtResponseRaw[0];
+      const useBanId = config?.useBanId;
       if (!useBanId) {
-        const message = `District cog ${cog} do not support BanID`;
-        responseBody = {
-          message,
-        };
-        console.log(`[${localCurrentDate()}] ${message}`);
+        throw new Error(`District id ${id} do not support BanID`);
         // TODO: Build Exploitation BDD (Legacy) by Legacy compose
       } else {
         const revision = await getRevisionFromDistrictCOG(cog);
         const revisionFileText = await getRevisionFileText(revision._id);
 
         // Update District with revision data
         const districtUpdate = {
-          id: district.id,
+          id,
           meta: {
-            ...district.meta,
-            bal:{
-              ...district.meta?.bal,
+            ...meta,
+            bal: {
+              ...meta?.bal,
               idRevision: revision._id,
               dateRevision: revision.publishedAt,
-            }
-          }
-        }
-        await partialUpdateDistricts([districtUpdate])
-        
+            },
+          },
+        };
+        await partialUpdateDistricts([districtUpdate]);
+
         responseBody = (await sendBalToBan(revisionFileText)) || {};
-        console.log(
-          `[${localCurrentDate()}] District cog ${cog} update in BAN BDD`
-        );
+        logger.info(`District id ${id} update in BAN BDD`);
+        logger.info(`Response body : ${JSON.stringify(responseBody)}`);
```

Comment:
Ça fonctionnerais de le mettre sur un seul appel ? (Voir même sur un seul 'Template String')
```suggestion
        logger.info(
          `District id ${id} update in BAN BDD`,
          `Response body : ${JSON.stringify(responseBody)}`,
        );
```